aws/session/session.go

```diff
@@ -395,7 +395,13 @@ func newSession(opts Options, envCfg envConfig, cfgs ...*aws.Config) (*Session,
 	// Load additional config from file(s)
 	sharedCfg, err := loadSharedConfig(envCfg.Profile, cfgFiles, envCfg.EnableSharedConfig)
 	if err != nil {
-		if _, ok := err.(SharedConfigProfileNotExistsError); !ok {
+		if len(envCfg.Profile) == 0 && !envCfg.EnableSharedConfig && envCfg.Creds.HasKeys() {
```

Comment:
This probably should look beyond envConfig, and include `aws.Config.Credentials` as well.